numpy/core/tests/test_ufunc.py

```diff
@@ -1936,3 +1936,31 @@ def test_nat_is_not_inf(self, nat):
             assert not np.isinf(nat)
         except TypeError:
             pass  # ok, just not implemented
+
+
+@pytest.mark.parametrize('ufunc', [getattr(np, x) for x in dir(np)
+                                if isinstance(getattr(np, x), np.ufunc)])
+def test_ufunc_types(ufunc):
+    '''
+    Check all ufuncs that the correct type is returned. Avoid
+    object and boolean types since many operations are not defined for
+    for them.
+
+    Choose the shape so even dot and matmul will succeed
+    '''
+    for typ in ufunc.types:
+        # types is a list of strings like ii->i
+        if 'O' in typ or '?' in typ:
+            continue
+        inp, out = typ.split('->')
+        args = [np.ones((3, 3), t) for t in inp]
+        with warnings.catch_warnings(record=True):
```

Comment:
What happens to all the recorded warnings?

Reply:
Could get the same effect by just ignoring them :)

Reply:
made the ignore explicit in the next line. Without `record=True` the test passes but all the warnings are printed out

Reply:
Shouldn't be printed out if they are ignored.

Reply:
hmm, filterwarnings('ignore') is not supported by pytest. @charris what change are you suggesting?

Reply:
Need to use `suppress_warnings`. @seberg Do you know if we still need that for Python >= 3.5?

Reply:
The failing test is one of ours that looks for "ignore" because using that used to mess up the warnings module.

Reply:
The code in this PR follows the pattern in [other places in the tests](https://github.com/numpy/numpy/blob/v1.16.3/numpy/core/tests/test_multiarray.py#L6980). Maybe we can accept this PR as-is and explore best practices for warnings in a separate issue/PR

Reply:
Catch warnings is OK, it just catches all warnings, which is not ideal (and `suppress_warnings` can be nested and works reasonable then). Should double check, but the doc I put on `suppress_warnings` indeed suggests that `"ignore"` filters should be fine in numpy 1.17. I remember there were still some oddities, but I do not think they were relevant for this.

Reply:
@seberg Sounds like we can disable the test and clean things up after we drop 1.16 support.

Reply:
@charris do you mean the "no ignore warnings test"? This here should work in any case, maybe not the prettiest, but `suppress_warnings` is also not perfect.